### PR TITLE
Remove ImmutableRegisterState and replace {get,set}_value in State with current_frame{,_mut}

### PR DESCRIPTION
**Pull request by bjorn3**
Based on https://github.com/bytecodealliance/wasmtime/pull/6178

### CODE REVIEW

**Comment by afonso360** [comment]
I might not be able to get to this today, but I'll look at it tomorrow.

File: cranelift/interpreter/src/state.rs [hunk 1]
```diff
@@ -26,29 +27,21 @@ pub trait State<'a> {
...
-    /// virtual register file.
-    fn get_value(&self, name: Value) -> Option<DataValue>;
-    /// Assign a value `V` to its [value reference](cranelift_codegen::ir::Value) in the
-    /// virtual register file.
-    fn set_value(&mut self, name: Value, value: DataValue) -> Option<DataValue>;
+    fn current_frame_mut(&mut self) -> &mut Frame<'a>;
+    fn current_frame(&self) -> &Frame<'a>;
+
     /// Collect a list of values `V` by their  [value references](cranelift_codegen::ir::Value);
     /// this is a convenience method for `get_value`. If no value is found for a value reference,
```
Comment (afonso360) [inline cranelift/interpreter/src/state.rs:40]: I think this comment might need an update now that we no longer have `get_value`

**Review by afonso360 — APPROVED** [review]
This seems like a nice simplification!

There were only 3 uses of `get_value`, but it looks like that interface was causing some unnecessary noise. Thanks!

**Comment by bjorn3** [comment]
Rebased

**Pull request merged**